test/test_distributed.py

```diff
@@ -0,0 +1,65 @@
+# Owner(s): ["oncall: distributed"]
+import contextlib
+import copy
+import functools
+import unittest
+from typing import Iterable, List, Tuple, Type, Union
+import torch
+import torch.distributed as dist
+import torch.nn as nn
+from torch.distributed._composable.fsdp import (
+    fully_shard,
+    OffloadPolicy,
+)
+from torch.testing._internal.common_cuda import TEST_CUDA
+from torch.testing._internal.common_distributed import skip_if_lt_x_gpu
+from torch.testing._internal.common_fsdp import (
+    FSDPTest,
+    FSDPTestMultiThread,
+    MLP,
+)
+from torch.testing._internal.common_utils import (
+    run_tests,
+)
+class TestFullyShard1DTrainingCore(FSDPTestMultiThread):
```

Comment:
this is great! a good next step is changing it to `TestFullyShard1DTrainingCore(FSDPTest)` so it's testing N GPUs for real. FSDPTestMultiThread runs on single GPU

Reply:
alright pushed this change now